Panic in product-controller

I got the following logs when the controller crashed

```bash
⋊> ~ ◦ k logs pod/mpas-product-controller-6f66cdf87-kwnsh -n mpas-system --previous                                                                              21:06:52
2023-09-13T19:05:44Z	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
2023-09-13T19:05:44Z	INFO	setup	starting manager
2023-09-13T19:05:44Z	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
2023-09-13T19:05:44Z	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
W0913 19:05:44.125995       1 warnings.go:70] v1beta2 GitRepository is deprecated, upgrade to v1
W0913 19:05:44.130382       1 warnings.go:70] v1beta2 GitRepository is deprecated, upgrade to v1
I0913 19:05:44.224383       1 leaderelection.go:248] attempting to acquire leader lease mpas-system/b3469b71.ocm.software...
I0913 19:06:02.871043       1 leaderelection.go:258] successfully acquired lease mpas-system/b3469b71.ocm.software
2023-09-13T19:06:02Z	DEBUG	events	mpas-product-controller-6f66cdf87-kwnsh_7a7a2f42-3293-46a9-b3e2-4ebd70ff1bee became leader	{"type": "Normal", "object": {"kind":"Lease","namespace":"mpas-system","name":"b3469b71.ocm.software","uid":"6d2f46bc-0f23-4a8d-b00c-773836a1f5c0","apiVersion":"coordination.k8s.io/v1","resourceVersion":"129323"}, "reason": "LeaderElection"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "source": "kind source: *v1alpha1.ProductDeploymentGenerator"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "source": "kind source: *v1alpha1.ComponentSubscription"}
2023-09-13T19:06:02Z	INFO	Starting Controller	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "productdeploymentpipeline", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentPipeline", "source": "kind source: *v1alpha1.ProductDeploymentPipeline"}
2023-09-13T19:06:02Z	INFO	Starting Controller	{"controller": "productdeploymentpipeline", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentPipeline"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "productdeployment", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeployment", "source": "kind source: *v1alpha1.ProductDeployment"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "productdeployment", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeployment", "source": "kind source: *v1alpha1.ProductDeploymentPipeline"}
2023-09-13T19:06:02Z	INFO	Starting Controller	{"controller": "productdeployment", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeployment"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "validation", "controllerGroup": "mpas.ocm.software", "controllerKind": "Validation", "source": "kind source: *v1alpha1.Validation"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "validation", "controllerGroup": "mpas.ocm.software", "controllerKind": "Validation", "source": "kind source: *v1beta2.GitRepository"}
2023-09-13T19:06:02Z	INFO	Starting Controller	{"controller": "validation", "controllerGroup": "mpas.ocm.software", "controllerKind": "Validation"}
2023-09-13T19:06:02Z	INFO	Starting EventSource	{"controller": "productdeploymentpipeline", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentPipeline", "source": "kind source: *v1alpha1.ProductDeploymentPipeline"}
2023-09-13T19:06:02Z	INFO	Starting Controller	{"controller": "productdeploymentpipeline", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentPipeline"}
2023-09-13T19:06:02Z	INFO	Starting workers	{"controller": "validation", "controllerGroup": "mpas.ocm.software", "controllerKind": "Validation", "worker count": 1}
2023-09-13T19:06:02Z	INFO	Starting workers	{"controller": "productdeploymentpipeline", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentPipeline", "worker count": 1}
2023-09-13T19:06:02Z	INFO	Starting workers	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "worker count": 1}
2023-09-13T19:06:02Z	INFO	Starting workers	{"controller": "productdeployment", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeployment", "worker count": 1}
2023-09-13T19:06:02Z	INFO	Starting workers	{"controller": "productdeploymentpipeline", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentPipeline", "worker count": 1}
2023-09-13T19:06:02Z	LEVEL(-4)	starting product deployment generator loop	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5"}
2023-09-13T19:06:03Z	INFO	fetching component	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "component": {"name":"mpas.ocm.software/podinfo","version":"1.0.2","registry":{"url":"ghcr.io/souleb"}}}
2023-09-13T19:06:03Z	LEVEL(-4)	configuring service account credentials	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5"}
2023-09-13T19:06:03Z	LEVEL(-4)	got service account	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "name": "mpas-podinfo-application"}
2023-09-13T19:06:05Z	INFO	retrieved component version, fetching ProductDescription resource	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "component": "mpas.ocm.software/podinfo"}
2023-09-13T19:06:05Z	INFO	fetched product description	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "description": {"name":"podinfo"}}
2023-09-13T19:06:05Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "config", "version": "1.0.0", "identities": [{"name":"backend"}], "identity": "\"name\"=\"config\""}
2023-09-13T19:06:07Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "instructions", "version": "1.0.0", "identities": [{"name":"backend"}], "identity": "\"name\"=\"instructions\""}
2023-09-13T19:06:09Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "validation", "version": "1.0.0", "identities": [{"name":"backend"}], "identity": "\"name\"=\"validation\""}
2023-09-13T19:06:11Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "config", "version": "1.0.0", "identities": [{"name":"frontend"}], "identity": "\"name\"=\"config\""}
2023-09-13T19:06:14Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "instructions", "version": "1.0.0", "identities": [{"name":"frontend"}], "identity": "\"name\"=\"instructions\""}
2023-09-13T19:06:16Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "validation", "version": "1.0.0", "identities": [{"name":"frontend"}], "identity": "\"name\"=\"validation\""}
2023-09-13T19:06:18Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "config", "version": "1.0.0", "identities": [{"name":"redis"}], "identity": "\"name\"=\"config\""}
2023-09-13T19:06:20Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "instructions", "version": "1.0.0", "identities": [{"name":"redis"}], "identity": "\"name\"=\"instructions\""}
2023-09-13T19:06:22Z	LEVEL(-4)	looking for resource data	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "resource": "validation", "version": "1.0.0", "identities": [{"name":"redis"}], "identity": "\"name\"=\"validation\""}
2023-09-13T19:06:24Z	INFO	successfully generated product deployment	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "productDeployment": {"name":"podinfo","namespace":"mpas-podinfo-application"}}
2023-09-13T19:06:24Z	LEVEL(-4)	snapshot-writer	creating snapshot for identity	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "identity": "\"component-name\"=\"mpas.ocm.software/podinfo\",\"component-version\"=\"1.0.2\",\"product-deployment-name\"=\"podinfo\""}
2023-09-13T19:06:24Z	LEVEL(-4)	snapshot-writer	built tar file	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5"}
2023-09-13T19:06:24Z	LEVEL(-4)	snapshot-writer	repository name constructed	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "name": "sha-7170694898901847774"}
2023-09-13T19:06:24Z	LEVEL(-4)	snapshot-writer	pushed data to the cache with digest	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "digest": "sha256:a6e2720a466b7d329a2400736cafc9fa514304a7747425b6c53850a6c6658aef"}
2023-09-13T19:06:24Z	INFO	snapshot-writer	snapshot successfully created/updated	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5", "digest": "sha256:a6e2720a466b7d329a2400736cafc9fa514304a7747425b6c53850a6c6658aef", "snapshot": {"namespace": "mpas-podinfo-application", "name": "podinfo-hbg3v4h"}}
2023-09-13T19:06:24Z	INFO	Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference	{"controller": "productdeploymentgenerator", "controllerGroup": "mpas.ocm.software", "controllerKind": "ProductDeploymentGenerator", "ProductDeploymentGenerator": {"name":"podinfo","namespace":"mpas-podinfo-application"}, "namespace": "mpas-podinfo-application", "name": "podinfo", "reconcileID": "76bf0a0f-d6e3-4732-8137-66296b72adb5"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x281a0f5]

goroutine 357 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x2b79ca0, 0x4fb1ae0})
	/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/panic.go:884 +0x213
github.com/open-component-model/mpas-product-controller/controllers.(*ProductDeploymentGeneratorReconciler).reconcile(0xc000933e00, {0x35b0500, 0xc000f4e330}, 0xc000f50340)
	/home/runner/work/mpas-product-controller/mpas-product-controller/controllers/productdeploymentgenerator_controller.go:316 +0x20b5
github.com/open-component-model/mpas-product-controller/controllers.(*ProductDeploymentGeneratorReconciler).Reconcile(0xc000933e00, {0x35b0500, 0xc000f4e330}, {{{0xc0008d67f8?, 0x0?}, {0xc000b97180?, 0x40e1c7?}}})
	/home/runner/work/mpas-product-controller/mpas-product-controller/controllers/productdeploymentgenerator_controller.go:173 +0x3b2
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x35b0500?, {0x35b0500?, 0xc000f4e330?}, {{{0xc0008d67f8?, 0x2a41540?}, {0xc000b97180?, 0x3596008?}}})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0005f5860, {0x35b0458, 0xc000933d60}, {0x2ca1840?, 0xc000dc8f40?})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0005f5860, {0x35b0458, 0xc000933d60})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:231 +0x587
```

--- a/controllers/productdeploymentgenerator_controller.go
+++ b/controllers/productdeploymentgenerator_controller.go
@@ -313,7 +313,7 @@ func (r *ProductDeploymentGeneratorReconciler) reconcile(ctx context.Context, ob
 	}
 
 	if project.Spec.Git.CommitTemplate == nil {
-		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.CommitTemplateEmptyReason, err.Error())
+		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.CommitTemplateEmptyReason, "commit template in project cannot be empty")
 
 		return ctrl.Result{}, fmt.Errorf("commit template in project cannot be empty")
 	}

